legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitlabWorkflowJobApi.java

```diff
@@ -0,0 +1,326 @@
+            {
+                throw new LegendSDLCServerException("Unknown workflow job in " + getRefInfoForException() + ", workflow " + workflowId + ": " + workflowJobId, Response.Status.NOT_FOUND);
+            }
+
+            return fromGitLabJob(this.projectId.toString(), job);
+        }
+
+        @Override
+        public List<WorkflowJob> getWorkflowJobs(String workflowId, Iterable<WorkflowJobStatus> statuses)
+        {
+            try
+            {
+                int pipelineId = parseIntegerIdIfNotNull(workflowId);
+                JobApi jobApi = getGitLabApi(this.projectId.getGitLabMode()).getJobApi();
+                List<Job> jobs = withRetries(() -> jobApi.getJobsForPipeline(this.projectId.getGitLabId(), pipelineId));
+                Stream<Job> jobStream = jobs.stream();
+
+                Set<WorkflowJobStatus> statusSet = (statuses == null) ? Collections.emptySet() : ((statuses instanceof Set) ? (Set<WorkflowJobStatus>) statuses : StreamSupport.stream(statuses.spliterator(), false).collect(Collectors.toCollection(() -> EnumSet.noneOf(WorkflowJobStatus.class))));
+                if (!statusSet.isEmpty())
+                {
+                    jobStream = jobStream.filter(job -> statusSet.contains(fromGitLabJobStatus(job.getStatus())));
+                }
+
+                return jobStream
+                        .map(p -> fromGitLabJob(this.projectId.toString(), p))
+                        .collect(Collectors.toList());
+            }
+            catch (Exception e)
+            {
+                throw buildException(e,
+                        () -> "User " + getCurrentUser() + " is not allowed to access workflow job for " + getRefInfoForException(),
+                        () -> "Unknown " + getRefInfoForException(),
+                        () -> "Error getting workflow job for " + getRefInfoForException());
+            }
+        }
+
+        @Override
+        public String getWorkflowJobsLogs(String workflowId, String workflowJobId)
+        {
+            int jobId = parseIntegerIdIfNotNull(workflowJobId);
+            Job job;
+            String jobTrace;
+            try
+            {
+                job = withRetries(() -> getGitLabApi(this.projectId.getGitLabMode()).getJobApi().getJob(this.projectId.getGitLabId(), jobId));
+                jobTrace = withRetries(() -> getGitLabApi(this.projectId.getGitLabMode()).getJobApi().getTrace(this.projectId.getGitLabId(), job.getId()));
+            }
+            catch (Exception e)
+            {
+                throw buildException(e,
+                        () -> "User " + getCurrentUser() + " is not allowed to access workflow job " + workflowJobId + " in " + getRefInfoForException(),
+                        () -> "Unknown workflow job in " + getRefInfoForException() + ": " + workflowJobId,
+                        () -> "Error getting workflow job " + workflowJobId + " in " + getRefInfoForException());
+            }
```

Comment:
Separate the error handling for getting the job and getting the trace. And make sure the error messages make clear whether the problem was with getting the job or with getting the job log.

Reply:
Handling errors for job and job log separately